### ISSUE TITLE
Remove direct dependency to openssl

### ISSUE THREAD

**Issue by axelfaure** [ThouCheese/cloud-storage-rs#77]
The library currently offers features allowing to choose whether to use rustls or openssl as a TLS backend. 
However, it also declares a direct, non optional dependency to openssl. If we keep this dependency, having a rustls feature doesn't really makes much sense. 

Openssl is also quite difficult to depend on when doing cross compilation (especially since cross does not support it any longer) so having the option to use this crate without openssl would be very helpful.

**Comment by ThouCheese** [comment]
You are right that this is not the most sensible thing. A PR to optionally replace OpenSSL is definitely useful and welcome!

**Comment by AlexanderThaller** [comment]
I'm having the same issue. I want to compile using musl which doesn't work with openssl.

From quick checking it seems that openssl is only used for creating message digests which could probably be replaced with native rust crates.

Would it be ok with you if I try to make the change and make a pull request?

**Comment by ThouCheese** [comment]
Sure!
